test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1

```diff
@@ -130,12 +136,14 @@ Describe 'Basic ThreadJob Tests' -Tags 'CI' {
         }
 
         $results = $job | Receive-Job -Wait
-        $results[0] | Should -Be $Var1
-        $results[1] | Should -Be $Var2
-        $results[2] | Should -Be $Var3
-        $results[3] | Should -Be 2
-        $results[4] | Should -Be $Var4
-        $results[5] | Should -Be $global:GVar1
+        $results | Should -BeExactly @(
+            $Var1
+            $Var2
+            $Var3
+            2
+            $Var4
+            $global:GVar1
+        )
```

Comment:
I think it is not correct. I can not approve such change.

I suggest close the PR.

Reply:
`Expected exactly @('Hello', 'Goodbye', 102, 2, 1, 2, 3, 4, 5, 'GlobalVar'), but got @('Hello', 'Goodbye', 102, 2, @(1, 2, 3, 4, 5), 'GlobalVar').`

Reply:
I found issues in Pester repository about arrays processing. So I'd don't want to spend time to investigate this because this does not add many value to our tests. I suggest close the PR and continue with others PRs.

Reply:
Array processing is typically only an issue when dealing with strongly-typed collections (e.g., KeyValueCollection) which aren't quite the same / don't implement all the same interfaces.

This specific example will fail due to pipeline unrolling of arrays; this should work here:
```ps1
        $results | Should -BeExactly @(
            $Var1
            $Var2
            $Var3
            2
            ,$Var4
            $global:GVar1
        )
```